lib/tasks/app_setup.rake

```diff
@@ -0,0 +1,20 @@
+namespace :app do
+  desc "Prepare Application on Boot Up"
+  task setup: :environment do
+    puts "\n== Preparing Elasticsearch =="
+    Rake::Task["search:setup"].execute
+
+    puts "\n== Preparing database =="
+    begin
+      Rake::Task["db:migrate"].execute
+    rescue ActiveRecord::NoDatabaseError
+      puts "\n== Creating and Seeding database =="
+      Rake::Task["db:create"].execute
```

Comment:
`Rake::Task["db:setup"].execute` for some reason does not work here. I had to break it down and explicitly call each task that it executes.

Reply:
Don't know exactly what the difference is, it might be because `db:setup` is just a reference to other 3 tasks run inline. You can use `system!()` here I guess.

Also to be forward looking we might want to just clone the tasks `db:setup` runs:

https://github.com/rails/rails/blob/7b5cc5a5dfcf38522be0a4b5daa97c5b2ba26c20/activerecord/lib/active_record/railties/databases.rake#L188-L189

My preference goes to `system()` though, as it allows us to pick up Rails changes. If that doesn't work with Docker, we can stick with the list of tasks in the Rails code

Reply:
I'll play around with this